### ISSUE TITLE
pnp typescript not compatible with SharePoint 2016 onwards, including 2019 and SharePoint Online

### ISSUE THREAD

**Issue by EA12** [pnp/pnpjs#749]
### Category
- [ ] Enhancement
- [x] Bug
- [ ] Question
- [ ] Documentation gap/issue

### Version

@pnp/odata@1.3.3
@pnp/sp@1.3.3
@pnp/logging@1.3.3
@pnp/common@1.3.3

@microsoft/generator-sharepoint@1.8.2
Build tools version: 2.5.3
Node version: v8.10.0

@microsoft/sp-build-web@1.1.0
   @microsoft/gulp-core-build-typescript@3.1.1
     @microsoft/api-extractor@2.3.8
    |  typescript@2.4.2
     typescript@2.2.2

### Expected / Desired Behavior / Question
gulp build should succeed.

### Observed Behavior
gulp build leads to many errors regarding pnp:
Error - typescript - node_modules@pnp\common\src\util.d.ts(66,56): error TS1005: ',' expected.
Error - typescript - node_modules@pnp\odata\src\queryable.d.ts(88,24): error TS1005: ',' expected.
Error - typescript - node_modules@pnp\sp\src\clientsidepages.d.ts(102,49): error TS1005: ',' expected.

### Steps to Reproduce
Use latest Yeomen generator 1.8.2
Add a new WebPart Project with yo @microsoft/sharepoint
Use "SharePoint 2016 onwards, including 2019 and SharePoint Online" as package.
Use "React" as Framework.
Install @pnp with npm install @pnp/logging @pnp/common @pnp/odata @pnp/sp --save
Add a import to the webpart file like so import { sp, SPRest } from '@pnp/sp';
Run gulp build

**Comment by koltyakov** [comment]
This is a well-known nuance and nothing can be changed on the library side. In SPFx for 2016 the old version of TypeScript is used. The only solution is to manually force usage of up to date TypeScript. One of the approaches is described in the docs https://pnp.github.io/pnpjs/documentation/SPFx-On-Premesis-2016/

**Comment by EA12** [comment]
The link you provided is about SPFx 1.4.1. not 1.8.2.
And has: ….SharePoint Framework projects targetting on-premesis **only**
As I wrote the package I choosed was ….and SharePoint Online...
Ok, that is may be due to an non updated documentation.

Removing modules by rm -rf node_modules did not work because -rf cannot be found as a parametername.
I removed the node_Modules folder manually and run npm Install.
After that, building the webpart succeeds, so the mentioned Workaround works. - Thanks.
